sormas-app/app/src/main/java/de/symeda/sormas/app/component/dialog/SynchronizationDialog.java

```diff
@@ -0,0 +1,589 @@
+			showPullInfrastructureProgressItems(false);
+			break;
+		case PULL_INFRASTRUCTURE_INITIAL:
+			showPullInfrastructureProgressItems(true);
+			break;
+		case CLEAR_UP:
+			showClearUpProgressItems(true);
+			break;
+		case SYNCHRONIZE:
+			showSynchronizeProgressItems();
+			break;
+		case CLEAR_UP_INFRASTRUCTURE:
+			showClearUpInfrastructureProgressItems(true);
+			break;
+		case DELETE_OBSOLETE:
+			showDeleteObsoleteProgressItems();
+			break;
+		case PULL_MODIFIED:
+			showPullModifiedProgressItems();
+			break;
+		case REPULL:
+			showRepullProgressItems();
+			break;
+		default:
+			throw new IllegalArgumentException(synchronizationStep.name());
+		}
+
+		getActivity().runOnUiThread(() -> {
+			contentBinding.stepsLayout.setCurrentStep(++currentSyncStepIndex);
+			contentBinding.stepsLayout.stepText
+				.setText(String.format(getActivity().getResources().getString(synchronizationStep.getCaptionResource()), currentSyncStepIndex));
+		});
+	}
+
+	private void updateStepNumber(int stepNumber) {
+		getActivity().runOnUiThread(() -> contentBinding.stepsLayout.setStepNumber(stepNumber));
+	}
+
+	private void loadNext() {
+
+		waitForUiThread();
+
+		currentBinding.setActive(false);
+		currentBinding.setDone(true);
+
+		if (progressItemBindings.size() > ++progressItemBindingsIndex) {
+			currentBinding = progressItemBindings.get(progressItemBindingsIndex);
+			currentBinding.setActive(true);
+		}
+	}
+
+	private void updatePulls(int pulled) {
+		waitForUiThread();
+		currentBinding.setPulls(currentBinding.getPulls() + pulled);
+	}
+
+	private void updatePushes(int pushed) {
+		waitForUiThread();
+		currentBinding.setPushes(currentBinding.getPushes() + pushed);
+	}
+
+	private void updatePushTotal(int pushTotal) {
+		waitForUiThread();
+		currentBinding.setPushTotal(pushTotal);
+	}
+
+	private void updateDeletions(int deleted) {
+		waitForUiThread();
+		currentBinding.setDeletions(currentBinding.getDeletions() + deleted);
+	}
+
+	private void showNextCleanupItems() {
+
+		if (currentSyncStep == SynchronizationStep.CLEAR_UP_INFRASTRUCTURE) {
+			showClearUpInfrastructureProgressItems(false);
+		} else if (currentSyncStep == SynchronizationStep.CLEAR_UP) {
+			showClearUpProgressItems(false);
+		} else {
+			throw new UnsupportedOperationException("Current sync step is no clear up step");
+		}
+	}
+
+	private void showProgressItems(boolean showPulls, boolean showPushes, List<String> captions) {
+
+		showProgressItems(showPulls, showPushes, false, captions);
+	}
+
+	private void showProgressItems(boolean showPulls, boolean showPushes, boolean showDeletions, List<String> captions) {
+
+		getActivity().runOnUiThread(() -> {
+			clearBindings();
+			LayoutInflater inflater = (LayoutInflater) getActivity().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+
+			captions.forEach(caption -> progressItemBindings.add(createBinding(inflater, caption, showPulls, showPushes, showDeletions)));
+
+			currentBinding = progressItemBindings.get(0);
```

Comment:
Please move these two calls into the synchronized block. I now it's not relevant in the current situation, but since the wait logic depends on the value of currentBinding, this should be within the block.